063-pdb-generation-environment-variable.md

```diff
@@ -0,0 +1,48 @@
+
+## Rationale
+
+- **Operational Constraints:** In environments where the creation of Pod Disruption Budgets (PDBs) is not permitted, users face significant challenges in deploying Strimzi effectively. This limitation can hinder the adoption and utility of Strimzi in such environments.
+
+- **Configuration Flexibility:** Users require the flexibility to configure their Strimzi deployments in accordance with their specific operational policies and constraints. A rigid approach that mandates the creation of PDBs may not be compatible with all operational environments.
+
+- **Consistency with Existing Features:** Implementing this feature as an environment variable aligns with the existing approach used for Network Policy Generation in Strimzi. This consistency simplifies the understanding and adoption of the feature by existing users.
+
+- **Simplicity and Ease of Use:** Offering a global setting to disable PDB generation avoids the complexity and repetitive configuration that would be required if this setting was to be managed at the individual custom resource level.
+
+## Current Implementation
+
+The Strimzi operator automatically creates PDBs to ensure high availability and minimize disruptions. However, this feature does not accommodate environments where PDB creation is restricted.
+
+## Proposal Details
+
+1. **Environment Variable Introduction:** Implement `STRIMZI_POD_DISRUPTION_BUDGET_GENERATION` with a default value of `true`.
+2. **Disabling PDB Generation:** When set to `false`, the operator will skip PDB operations for all Strimzi components. It will not create, modify or delete any PDB.
+3. **Documentation and Guidance:** Update Strimzi documentation to include instructions and implications of disabling PDB generation.
+4. **Scope:** This change affects only the Pod Disruption Budget aspect and does not alter any other functionalities of the Strimzi operator.
+
+## Affected Projects
+
+This proposal pertains solely to the [Strimzi Kafka Operator](https://github.com/strimzi/strimzi-kafka-operator).
+
+## Compatibility
+
+- **Default Behavior:** By retaining `true` as the default value, existing deployments remain unaffected.
+- **Backward Compatibility:** This feature is an addition and does not alter existing functionalities.
+
+## Rejected Alternatives
+
+1. **CRD-based Option for PDB Disabling:** Initially considered, this was rejected for increasing complexity and requiring repetitive configuration.
+2. **Global Disabling via Command-Line Flag:** Although an alternative, a command-line flag was deemed less flexible and consistent compared to an environment variable.
```

Comment:
Curious about this one, as we don't have any command line tool. Do you mean when using Helm as deployment method?

Reply:
no I meant as a flag in entrypoint command of strimzi image

Reply:
@fvaleri done